tsl/src/nodes/gapfill/gapfill_exec.c

```diff
@@ -656,10 +656,32 @@ gapfill_advance_timestamp(GapFillState *state)
 			 * To be consistent with time_bucket we do UTC bucketing unless
 			 * a different timezone got explicity passed to the function.
 			 */
-			next = DirectFunctionCall2(state->have_timezone ? timestamptz_pl_interval :
-															  timestamp_pl_interval,
-									   TimestampTzGetDatum(state->gapfill_start),
-									   IntervalPGetDatum(state->next_offset));
+			if (state->have_timezone)
+			{
+				bool isnull;
+				/* TODO: optimize by constifying and caching the datum if possible */
+				Datum tzname = gapfill_exec_expr(state, get_timezone_arg(state), &isnull);
```

Comment:
Opted for simple approach, we can optimize the performance if necessary later on.

Reply:
What is needed for caching, something more than saving this datum in the GapFillState?